### ISSUE TITLE
TRestStringOutput sometimes causes crash at the end of a simulation

### ISSUE THREAD

**Issue by lobis** [rest-for-physics/framework#23]
**Edit**: Thanks to @nkx111 the issue was found in `TRestStringOutput`, probably it was related to resizing the console due to using screen. The issue is now solved in commit f8a2e8bf9bc634a98896c29ce5c602f523fc630d.

I found out that if I don't set the `runNumber` to `auto` I sometimes get a crash at the end of the simulation. Here are the relevant lines in the output with debug enabled.

```
DEBUG: Energy deposited in SENSITIVE volume: 0 keV
DEBUG: Events depositing energy in sensitive volume: 36713/10000
DEBUG: End of event ID 9999 (10000 of 10000)

=======================================================================
======       Pre-compound/De-excitation Physics Parameters     ========
=======================================================================
Type of pre-compound inverse x-section              3
Pre-compound model active                           1
Pre-compound low energy (MeV)                       0.1
Type of de-excitation inverse x-section             3
Type of de-excitation factory                       Evaporation
Number of de-excitation channels                    8
Min excitation energy (keV)                         0.01
Min energy per nucleon for multifragmentation (MeV) 1e+05
Level density (1/MeV)                               0.1
Time limit for long lived isomeres (ns)             1442.7
Internal e- conversion flag                         1
Store e- internal conversion data                   1
Electron internal conversion ID                     0
Correlated gamma emission flag                      0
Max 2J for sampling of angular correlations         10
=======================================================================
======================== run summary ======================
10000 Events simulated
===========================================================
Graphics systems deleted.
Visualization Manager deleting...
-- Debug : TRestRun::WriteWithDataBase. Getting entries in analysisTree
-- Debug : TRestRun::WriteWithDataBase. Entries found : 36713
-- Debug : TRestRun::WriteWithDataBase. Calling this->Write(0,kWriteDelete)
-- Debug : TRestRun::WriteWithDataBase. Succeed
-- Debug : TRestRun::WriteWithDataBase. fMetadata.size() == 2
-- Debug : TRestRun::WriteWithDataBase. fInputMetadata.size() == 0
-- Debug : NO historic
-- Debug : TRestRun::WriteWithDataBase. fInputMetadata.size() == 0
-- Debug : NO historic
-- Debug : TResRun::WriteWithDataBase. Run number is : 0
terminate called after throwing an instance of 'std::length_error'
  what():  basic_string::_M_create
CosmicParticles.sh: línea 28: 31047 Abortado                restG4 CosmicParticles.rml
```

I cannot reproduce this issue though, I tried doing a shorter simulation also setting the `runNumber` to 0 but this time everything went fine.

Based on the output log I think the problem occurs in line 951 of `TRestRun.cxx`, in the block

 ```
 // write to database
    debug << "TResRun::WriteWithDataBase. Run number is : " << fRunNumber << endl;
    if (fRunNumber != -1) {
        int fileid = gDataBase->set_runfile(fRunNumber, (string)fOutputFileName);
        fout << "DataBase Entry Added! Run Number: " << fRunNumber << ", File ID: " << fileid << endl;
    }
```

**Comment by lobis** [comment]
```
                                            ||                      ++ g4Neutrons_vetoWestMiddle3of3VolumeEDepQf0_150000                      ||                        
                                            ||                      ++ g4Neutrons_vetoWestMiddle3of3VolumeEDepQf1_000000                      ||                        
                                            ||                        +++++++++++++++++++++++++++++++++++++++++++++++                         ||                        
                                            ||                                                                                                ||                        
                                            ||                                                                                                ||                        
                                            ====================================================================================================                        

                                            [==                            ++ TRestGeant4AnalysisProcess : g4Ana                             ==]                        
                                            [==                          ++ TRestGeant4VetoAnalysisProcess : g4Veto                          ==]                        
                                            [==                ++ TRestGeant4CosmicNeutronTaggingAnalysisProcess : g4Neutrons                ==]                        
                                            ====================================================================================================                        
                                            [==                          TRestProcessRunner: Starting the Process..                          ==]                        
-- Info : TRestThread : Writing temp file. Thread id : 0                        terminate called after throwing an instance of 'std::length_error'--------------  what():  basic_string::_M_create
```

I had this error occur during an analysis, of a file that had the runNumber set by auto, and I can read it to be 2534, so perhaps the runNumber is not the problem.

```
root [1] EventTree->Show(0)
======> EVENT:0
 TRestGeant4EventBranch = (TRestGeant4Event*)0x55d2b0848490
 fUniqueID       = 0
 fBits           = 50331648
 fRunOrigin      = 2534
 fSubRunOrigin   = 0
 fEventID        = 0
 fSubEventID     = 0
 fSubEventTag    =
 fEventTime.fSec = 0
 fEventTime.fNanoSec = 0
 fOk             = 1
 PerProcessEnergyInitFlag = 0
 fPrimaryEventOrigin = TVector3
 fPrimaryParticleName = (vector<TString>*)0x55d2b0848668
 fPrimaryEventDirection = (vector<TVector3>*)0x55d2b0848680
 fPrimaryEventEnergy = (vector<double>*)0x55d2b0848698
 fTotalDepositedEnergy = 12.832
 fSensitiveVolumeEnergy = 0
 fNVolumes       = 110
 fVolumeStored   = (vector<int>*)0x55d2b08486c8
 fVolumeStoredNames = (vector<string>*)0x55d2b08486e0
 fVolumeDepositedEnergy = (vector<double>*)0x55d2b08486f8
 fNTracks        = 1
 fTrack          = (vector<TRestGeant4Track>*)0x55d2b08489a0
 fTrack.fUniqueID = 0
 fTrack.fBits    = 33554432
 fTrack.fTrack_ID = 1
 fTrack.fParent_ID = 0
 fTrack.fSubEventId = 0
 fTrack.fParticleName = neutron
 fTrack.fGlobalTimestamp = 0
 fTrack.fTrackTimestamp = 0.0243238
 fTrack.fKineticEnergy = 89116.8
 fTrack.fHits.fUniqueID = 0
 fTrack.fHits.fBits = 33554432
 fTrack.fHits.fNHits = 58
 fTrack.fHits.fTotEnergy = 12.832
 fTrack.fHits.fX = (vector<float>*)0x55d2b09fc370
 fTrack.fHits.fY = (vector<float>*)0x55d2b09fc388
 fTrack.fHits.fZ = (vector<float>*)0x55d2b09fc3a0
 fTrack.fHits.fT = (vector<float>*)0x55d2b09fc3b8
 fTrack.fHits.fEnergy = (vector<float>*)0x55d2b09fc3d0
 fTrack.fHits.fType = (vector<REST_HitType>*)0x55d2b09fc3e8
 fTrack.fHits.fVolumeID = printing kAny case (62)
 fTrack.fHits.fProcessID = printing kAny case (62)
 fTrack.fHits.fKineticEnergy = printing kAny case (62)
 fTrack.fHits.fMomentumDirectionX = printing kAny case (62)
 fTrack.fHits.fMomentumDirectionY = printing kAny case (62)
 fTrack.fHits.fMomentumDirectionZ = printing kAny case (62)
 fTrack.fTrackOrigin = TVector3
 fMaxSubEventID  = 0
root [2]
```

**Comment by jgalan** [comment]
It could be possible that the origin of the problem is common, but in principle, one is happening with `restG4` execution, the other with `restManager` execution, it looks as if they are unrelated errors? 

For the second case (restManager), what it happens if you leave only the `TRestGeant4AnalysisProcess`?

**Comment by lobis** [comment]
> For the second case (restManager), what it happens if you leave only the `TRestGeant4AnalysisProcess`?

It happened the same as before, I could not reproduce the crash, I tried lowering the number of events to be processed first (because it takes a while to do them all) and it worked just fine. I changed the number of events and rerun the command a few times without issues.

It looks like it has something to do this this `database` (which I don't know what it is), as if the first time I launch the command its trying to retrieve some value that does not exists and fails, but after the first time it doesn't happen again.

**Comment by nkx111** [comment]
You mean the problem only occurs on the fresh install and first running of REST?

**Comment by nkx111** [comment]
I think I find the issue, it is because the console resized(narrowed) during the process. I successfully reproduce the error with this operation. 

So the bug lies in `TRestStringOutput` instead of `database`. I have fixed it in commit f8a2e8bf9bc634a98896c29ce5c602f523fc630d
